### PR TITLE
Fixes #6229 - tripwire mines not working / brings back frikin' laser beams

### LINKED ISSUE #6229: Tripwire Mines Do Not Work
Infrared assemblies work as they should but using them in a grenade casing does not. No matter what option you set in the assembly menu, before or after attaching it to grenade case, it will never emit the laser required to trigger the assembly/bomb.

### PR DESCRIPTION
**Pull request by Kyep**
Fixes & Tweaks Infrared Lasers:
- Infrared laser assemblies now actually work as device (ie: bomb) triggers. Fixes #6229
- This means you can once again set up laser mines that detonate when their beam is crossed.
- Infrared laser assemblies now default to emitting visible red lasers. You can still toggle them to emitting invisible lasers.
- Some pre-made explosive tripwire mines have been added. /obj/item/weapon/grenade/chem_grenade/waterpotassium/tripwire, tripwire_armed, and tripwire_armed_stealth. The first is safe to handle. The second arms itself on spawn. The third arms itself on spawn, and uses an invisible laser. All three contain 100 water/potassium, and will probably blow your limbs off if they go off close to you.
- The pre-armed versions are intended for use in mapping. If you place them in a map and set their 'dir' variable, they will project lasers in that direction a few seconds after the map loads.
- All tripwire mines will occasionally (every 20 cycles) refresh (recreate) their lasers. This addresses cases where a tripwire mine is moved in an unexpected fashion, e.g: teleported, destroyed, etc. It prevents the lasers from hanging around in incorrect locations.
- As a positive side effect of this, mines will also blink off for one cycle every 20 cycles. This means that if you're well positioned and act fast, it is possible to get past a laser minefield using manual dexterity.

Semi-unrelated fixes/tweaks:
- Some types of chem grenades in custom_grenades have been changed from relative to absolute pathing. IE: "New()" to "/obj/item/weapon/grenade/chem_grenade/lube/remote/New()". I was editing that area anyway, so it made sense to put in absolute paths for their procs.

Example of pre-made, visible laser traps:
infrared_laser_looks

### CODE REVIEW

**Comment by Shadeykins** [comment]
Not sure if it's possible, but higher transparency lasers may look better.

**Comment by Kyep** [comment]
The red lasers are intended to stand out as much as possible. They're mostly there so you can experiment with the beams and understand how they work.

If you want stealth, then toggle the infrared emitter over to 'invisible' mode. Then you cannot see the beams at all.

**Comment by Tayyyyyyy** [comment]
I feel like they look really similar to cable wiring right now, though. A tiny bit of transparency might help with possible confusion, plus it'd look cooler.

**Comment by Ralta** [comment]
I agree they should be super visible, but I also agree with @Tayyyyyyy that they look rather like cabling at the moment. I feel like a degree of transparency would, as @Tayyyyyyy said, make the difference more distinctive.

Either way, +1 from me.

**Comment by Jountax** [comment]
It was still possible to make tripwire bombs, you just had to wire the laser to a signaller that would transmit to a signaller attached to the bomb. I welcome the behavior being fixed, however.

**Comment by LightFire53** [comment]
Dynamite with a laser beam, anyone?

**Comment by Kyep** [comment]
Closing & Reopening to force Travis recheck.

**Comment by Kyep** [comment]
For the record, I'm not inclined to change the look of the lasers. Cables don't run on top of floors. Lasers do. Cables also don't blink on/off. These do. They should be distinguishable. Also, I am not a spriter. If someone else wants to take on the task of making them look fancy, great, but that's not really my area. I just want to fix the functionality so they actually work.

**Comment by taukausanake** [comment]
I think the only issue I have with the lasers (aside from them being pretty thicc) is that the North and South icons aren't shooting out from the laser pointer. That would be more work than needed though but _muh immersions_

**Comment by Kyep** [comment]
No graphical tweaks/updates for them are going to be considered in this PR.
This PR is simply about making them *work*.
IE: making sure that when you turn one on, it actually projects a beam, and can be activated by a mob crossing the beam. 
That is it. Just the absolutely essential functionality required for them to be usable when constructed or pre-placed on a map.

Any graphical updates will have to be a separate PR.

File: code/modules/assembly/infrared.dm [hunk 1]
```diff
@@ -36,11 +44,22 @@
...
-		if(first)	qdel(first)
+		if(first)
+			qdel(first)
 		processing_objects.Remove(src)
 	update_icon()
 	return secured
 
+/obj/item/device/assembly/infra/New()
+	..()
+	if (!secured)
```
Comment (tigercat2000) [inline code/modules/assembly/infrared.dm]: No spaces in control statements

File: code/modules/assembly/infrared.dm [hunk 2]
```diff
@@ -36,11 +44,22 @@
...
+		toggle_secure()
+
+/obj/item/device/assembly/infra/proc/arm() // Forces the device to arm no matter its current state.
+	if (!secured) // Checked because arm() might be called sometime after the object is spawned.
```
Comment (tigercat2000) [inline code/modules/assembly/infrared.dm]: No spaces in control statements

File: code/modules/assembly/infrared.dm [hunk 3]
```diff
@@ -22,7 +26,11 @@
 	return ..()
 
 /obj/item/device/assembly/infra/describe()
-	return "The infrared trigger is [on?"on":"off"]."
+	return "The assembly is [secured?"secure":"not secure"]. The infrared trigger is [on?"on":"off"]."
```
Comment (tigercat2000) [inline code/modules/assembly/infrared.dm]: Spacing between `?` and `:` operators, `secured ? "secure" : "not secure"`

**Comment by Kyep** [comment]
@tigercat2000 All your change requests are done.

**Comment by Ralta** [comment]
@Kyep That's fair - glad to see they're getting fixed anyway! And now that the discussion has been had, I'm sure someone will make a PR to adjust the lazy-beams 😄

**Comment by DarkPyrolord** [comment]
👍

**Pull request merged**